packages/lib/contracts/application/BaseApp.sol

```diff
@@ -11,17 +10,16 @@ import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
  * It handles the creation and upgrading of proxies.
  */
 contract BaseApp is Ownable {
-  /// @dev Factory that creates proxies.
-  UpgradeabilityProxyFactory public factory;
+  /**
+   * @dev Emitted when a new proxy is created.
+   * @param proxy Address of the created proxy.
+   */
+  event ProxyCreated(address indexed proxy);
 
   /**
    * @dev Constructor function
-   * @param _factory Proxy factory
    */
-  constructor(UpgradeabilityProxyFactory _factory) public {
-    require(address(_factory) != address(0), "Cannot set the proxy factory of an app to the zero address");
-    factory = _factory;
-  }
+  constructor() public {}
```

Comment:
I hesitated about entirely removing the ProxyFactory, since the original idea was that we could have a single ProxyFactory shared by all Apps, so each user would only need to deploy the App. However, since we are now aiming at making these contracts upgradeable, gas cost for the end user is exactly the same, since they only need to deploy a proxy to App, regardless of where the factory code for proxies is. So +1 to this change.